Dataset Load

In [1]:
import pandas as pd

df = pd.read_csv("Books_Data_Clean.csv")

df.head()

,index,Publishing Year,Book Name,Author,language_code,Author_Rating,Book_average_rating,Book_ratings_count,genre,gross sales,publisher revenue,sale price,sales rank,Publisher,units sold
0,0,1975.0,Beowulf,"Unknown, Seamus Heaney",en-US,Novice,3.42,155903,genre fiction,34160.0,20496.0,4.88,1,HarperCollins Publishers,7000
1,1,1987.0,Batman: Year One,"Frank Miller, David Mazzucchelli, Richmond Lew...",eng,Intermediate,4.23,145267,genre fiction,12437.5,7462.5,1.99,2,HarperCollins Publishers,6250
2,2,2015.0,Go Set a Watchman,Harper Lee,eng,Novice,3.31,138669,genre fiction,47795.0,28677.0,8.69,3,"Amazon Digital Services, Inc.",5500
3,3,2008.0,When You Are Engulfed in Flames,David Sedaris,en-US,Intermediate,4.04,150898,fiction,41250.0,24750.0,7.50,3,Hachette Book Group,5500
4,4,2011.0,Daughter of Smoke & Bone,Laini Taylor,eng,Intermediate,4.04,198283,genre fiction,37952.5,22771.5,7.99,4,Penguin Group (USA) LLC,4750


Q1: How many rows and columns does this dataset have?

In [2]:
df.shape

(1070, 15)

Q2: Which columns have missing values and how many are missing in each?

In [3]:
# df.isnull().sum()
df.isnull().sum()[df.isnull().sum() > 0]

Publishing Year     1
Book Name          23
language_code      53
dtype: int64

Q3: What is the average Book_average_rating across all books?

In [4]:
df["Book_average_rating"].mean()

np.float64(4.007)

Q4: What is the total gross sales across all 1070 books?

In [5]:
df["gross sales"].sum()

np.float64(1986586.5500000003)

Q5: Which genre appears most frequently?

In [6]:
df["genre"].value_counts()

genre
genre fiction    822
nonfiction       171
fiction           62
children          15
Name: count, dtype: int64

Q6: Which publisher has the highest total units sold?

In [9]:
df.columns = df.columns.str.strip()

publisher_units = df.groupby("Publisher")["units sold"].sum()

publisher_units.sort_values(ascending=False).head()

Publisher
Amazon Digital Services,  Inc.    6074136
Random House LLC                  1315958
Penguin Group (USA) LLC            934303
HarperCollins Publishers           563409
Hachette Book Group                543821
Name: units sold, dtype: int64

Q7: Find the correlation between gross sales and publisher revenue

In [12]:
correlation = df["gross sales"].corr(df["publisher revenue"])

print("Correlation:", correlation)

if abs(correlation) >= 0.7:
    print("Strong correlation")
else:
    print("Weak correlation")

Correlation: 0.9128912394213524
Strong correlation
